src/array_api_stubs/_draft/array_object.py

```diff
@@ -1,25 +1,28 @@
 from __future__ import annotations
 
-from ._types import (
-    array,
-    dtype as Dtype,
-    device as Device,
-    Optional,
-    Tuple,
-    Union,
-    Any,
-    PyCapsule,
-    Enum,
-    ellipsis,
-)
-
-
-class _array:
-    def __init__(self: array) -> None:
+from typing import TYPE_CHECKING, Protocol, TypeVar
+from enum import Enum
+from .data_types import DType
+from ._types import device as Device
+
+if TYPE_CHECKING:
+    from ._types import (
+        Any,
+        PyCapsule,
+        ellipsis,
+    )
+
+array = TypeVar("array", bound="Array")
+# NOTE: when working with py3.11+ this can be ``typing.Self``.
+
+
+class Array(Protocol):
+    def __init__(self) -> None:
         """Initialize the attributes for the array object class."""
+        ...
 
     @property
-    def dtype(self: array) -> Dtype:
+    def dtype(self) -> "DType":
```

Comment:
FYI you won't need any quotation marks if `from __future__ import annotations` is enabled as this (implicitly) stringifies all annotations anyway.

Reply:
This occurred while trying to get the docs to render. Many things were tested, so I don't know if this is strictly necessary (I'll check), but it was not related to type hint stringification.

Reply:
It's not necessary! Removed.